src/autoimport/model.py

```diff
@@ -37,12 +37,15 @@
 class SourceCode:  # noqa: R090
     """Python source code entity."""
 
-    def __init__(self, source_code: str) -> None:
+    def __init__(
+        self, source_code: str, config: Optional[Dict[str, Any]] = None
+    ) -> None:
         """Initialize the object."""
         self.header: List[str] = []
         self.imports: List[str] = []
         self.typing: List[str] = []
         self.code: List[str] = []
+        self.config: Dict[str, Any] = config if config else {}
```

Comment:
Shouldn't the type of `self.config` be `ProjectConfig`?

Reply:
If you agree with me, there are other functions where the typing could change

Reply:
The argument actually has type `Dict[str Any]`, because (in the `cli.py` file) I've called the `ProjectConfig.to_dict` method.

Quoting from `cli.py`:
```
     config = ProjectConfig(project_name="autoimport").to_dict()
```

My reason for converting the `ProjectConfig` object to a dictionary (instead of passing around the `ProjectConfig` object itself) was that I thought this would make this project (`autoimport`) less tightly-coupled to the types that are exported by `maison`. Using a dict instead of a `ProjectConfig` object also made it easier for me to write the tests for this feature.

Reply:
That makes total sense